portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/BullhornServiceImpl.java

```diff
@@ -0,0 +1,339 @@
+                                    , new Object[] {from,to,event,ref,e.getEventTime(),url}
+                                    , "ID");
+                            countCache.remove(to);
+                        } else if (ProfileConstants.EVENT_FRIEND_CONFIRM.equals(event)
+                                    || ProfileConstants.EVENT_FRIEND_IGNORE.equals(event)) {
+                            String to = pathParts[2];
+                            sqlService.dbWrite("DELETE FROM SOCIAL_ALERTS WHERE EVENT = ? AND FROM_USER = ?"
+                                    , new Object[] {ProfileConstants.EVENT_FRIEND_REQUEST, to});
+                            Session session = sessionManager.getCurrentSession();
+                            String oldUserId = session.getUserId();
+                            session.setUserId(to);
+                            String url = profileLinkLogic.getInternalDirectUrlToUserConnections();
+                            session.setUserId(oldUserId);
+                            sqlService.dbInsert(null
+                                    , SOCIAL_INSERT_SQL
+                                    , new Object[] {from,to,event,ref,e.getEventTime(),url}
+                                    , "ID");
+                            countCache.remove(from);
+                        } else if (ProfileConstants.EVENT_MESSAGE_SENT.equals(event)) {
+                            String to = pathParts[2];
+                            Session session = sessionManager.getCurrentSession();
+                            String oldUserId = session.getUserId();
+                            session.setUserId(to);
+                            String url = profileLinkLogic.getInternalDirectUrlToUserProfile(to);
+                            session.setUserId(oldUserId);
+                            sqlService.dbInsert(null
+                                    , SOCIAL_INSERT_SQL
+                                    , new Object[] {from,to,event,ref,e.getEventTime(),url}
+                                    , "ID");
+                            countCache.remove(to);
+                        } else if (ContentHostingService.EVENT_RESOURCE_ADD.equals(event)) {
+                            String siteId = pathParts[3];
+                            // In this case title = filename
+                            String title = pathParts[pathParts.length - 1];
+
+                            try {
+                                Site site = siteService.getSite(siteId);
+                                String toolId = site.getToolForCommonId("sakai.resources").getId();
+                                String url = serverConfigurationService.getPortalUrl() + "/site/"
+                                                                            + siteId + "/tool/" + toolId;
+                                // Get all the members of the site
+                                for (Member member : site.getMembers()) {
```

Comment:
Could use a stream and filter here instead of the for/if. Total sex appeal in your code.